slsDetectorServers/ctbDetectorServer/slsDetectorFunctionList.c

```diff
@@ -898,6 +898,24 @@ int getReadoutMode() {
 }
 
 /* parameters - timer */
+int setNextFrameNumber(uint64_t value) {
+    LOG(logINFO,
+        ("Setting next frame number: %llu\n", (long long unsigned int)value));
+    setU64BitReg(value, NEXT_FRAME_NUMB_LOCAL_LSB_REG,
+                 NEXT_FRAME_NUMB_LOCAL_MSB_REG);
+#ifndef VIRTUAL
+    // for 1g udp interface
+    setUDPFrameNumber(value);
+#endif
+    return OK;
+}
+
+int getNextFrameNumber(uint64_t *retval) {
```

Comment:
What will the virtual server return?

Reply:
what ever is in this register.. it is set up to be 1 at startup..

Reply:
the return is a ok or fail (useful only for eiger).. but the argument will return whatever is in the register for the next frame number.